src/app/shared/services/dynamic-data/actions/set_data.action.spec.ts

```diff
@@ -208,4 +212,93 @@ describe("set_data Action", () => {
       `[Update Fail] no doc exists\ndata_list: test_flow\n_index: 10`
     );
   });
+
+  /*************************************************************
+   *  mergeUpdatesById Tests
+   ************************************************************/
+  describe("mergeUpdatesById", () => {
+    it("merges updates for the same ID correctly", () => {
+      const updates = [
+        { id: "123", user: { name: "Jasper" } },
+        { id: "123", user: { age: 9 } },
+        { id: "123", status: "active" },
+        { id: "123", status: "inactive" },
```

Comment:
Question - How exactly would we end up in a case like this from authoring? I thought a `set_data` action already should include all the updates to apply to a specific id e.g. 
```
click | set_data | 
_list_id: feat_data_actions_list,
completed:true
number: 20
```
In this case yes there would be batch update for all items in the data_list, but each item should already be collated for the `completed: true, number:20` partial update? I'm confused how we end up with successive updates to the same ID.

In the case of multiple actions each would be processed in turn so also wouldn't be batched together

Reply:
Ah yes I think you're right. I got caught up in trying to optimise generally without thinking about what is actually possible through authoring. In which case, the performance gains I noticed, if genuine, must all be coming from the batching optimisation.